support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -0,0 +1,268 @@
+import { useEffect, useState } from 'react';
+import {
+	number,
+	object,
+	type Output,
+	parse,
+	picklist,
+	record,
+	string,
+} from 'valibot'; // 1.54 kB
+import { Box, BoxContents } from 'components/checkoutBox/checkoutBox';
+import { CheckoutHeading } from 'components/checkoutHeading/checkoutHeading';
+import { Header } from 'components/headers/simpleHeader/simpleHeader';
+import { ContributionsOrderSummary } from 'components/orderSummary/contributionsOrderSummary';
+import { PageScaffold } from 'components/page/pageScaffold';
+import { SecureTransactionIndicator } from 'components/secureTransactionIndicator/secureTransactionIndicator';
+import type { Currency } from 'helpers/internationalisation/currency';
+import { currencies } from 'helpers/internationalisation/currency';
+import { renderPage } from 'helpers/rendering/render';
+import { trackComponentClick } from 'helpers/tracking/behaviour';
+import { GuardianTsAndCs } from 'pages/supporter-plus-landing/components/guardianTsAndCs';
+
+const darkBackgroundContainerMobile = css`
+	background-color: ${palette.neutral[97]};
+	${until.tablet} {
+		background-color: ${palette.brand[400]};
+		border-bottom: 1px solid ${palette.brand[600]};
+	}
+`;
+
+const columns = css`
+	position: relative;
+	color: ${palette.neutral[7]};
+	${textSans.medium()};
+	padding-top: ${space[2]}px;
+`;
+
+const secureTransactionIndicator = css`
+	margin-bottom: ${space[3]}px;
+	${from.tablet} {
+		margin-bottom: ${space[4]}px;
+	}
+`;
+
+const shorterBoxMargin = css`
+	:not(:last-child) {
+		${until.tablet} {
+			margin-bottom: ${space[2]}px;
+		}
+	}
+`;
+
+const searchParams = new URLSearchParams(window.location.search);
+const query = {
+	productId: searchParams.get('product'),
+	ratePlanId: searchParams.get('ratePlan'),
+	chargeId: searchParams.get('charge'),
+};
+
+const countryGroupIds = ['uk', 'us', 'eu', 'au', 'nz', 'ca', 'int'] as const;
+const CountryGroupIdSchema = picklist(countryGroupIds);
+const countryGroupId = parse(
+	CountryGroupIdSchema,
+	window.location.pathname.split('/')[1],
+);
+
+let currentCurrency: Currency;
+let currentCurrencyKey: keyof typeof currencies;
+switch (countryGroupId) {
+	case 'uk':
+		currentCurrency = currencies.GBP;
+		currentCurrencyKey = 'GBP';
+		break;
+	case 'us':
+		currentCurrency = currencies.USD;
+		currentCurrencyKey = 'USD';
+		break;
+	case 'au':
+		currentCurrency = currencies.AUD;
+		currentCurrencyKey = 'AUD';
+		break;
+	case 'eu':
+		currentCurrency = currencies.EUR;
+		currentCurrencyKey = 'EUR';
+		break;
+	case 'nz':
+		currentCurrency = currencies.NZD;
+		currentCurrencyKey = 'NZD';
+		break;
+	case 'ca':
+		currentCurrency = currencies.CAD;
+		currentCurrencyKey = 'CAD';
+		break;
+}
+
+const ProductSchema = object({
+	ratePlans: record(
+		object({
+			id: string(),
+			charges: record(
+				object({
+					id: string(),
+					pricing: record(number()),
+				}),
+			),
+		}),
+	),
+});
```

Comment:
I have introduced [`valibot`](https://valibot.dev/) to the mix, perhaps prematurely?

It's similar to `zod`, but way smaller so generally better to include on the client. [There's an interesting article about it here](https://www.builder.io/blog/introducing-valibot).

I thought if we are to start including data from external sources, it's not a bad idea to parse it prior to trying to render it, allowing for better error boundaries.

We do parse all JSON in Scala and I like the confidence it gives me to know my data is in `X` shape.

Reply:
Ping @tjmw we I know you were interested in this.

Reply:
Hmm, having quite a few VStudio eslint complaints with `valibot` -  maybe can double check my setup next week? Can see how your loading the schema via useEffect into setProducts useState so can access within the Checkout.